### PR TITLE
Fix secondary nav when browser settings are set to large for font size

### PR DESCRIPTION
**Pull request by jzucadi**
## Summary

When a user has set the font six in their chrome browser to Large, this causes the secondary nav on the MHV landing page to display nav items in an unwanted state. The nav items get wrapped when they should appear on one line. This also occurs in Safari when a user has set the font size to 24.

This PR implements a font size detection mechanism for the MhvSecondaryNavMenu component. It uses try-catch blocks to prevent test failures. The changes include:

- A checkFontSize function using useCallback to detect large text settings.
- A useEffect hook to set up font size checking on mount and during resizes.
- Implementation of ResizeObserver with a fallback to window resize event.
- Error logging for easier debugging and graceful error handling.
- Detailed comments explaining the font size detection logic and error handling approach.

These changes aim to prevent the menu bar from rendering incorrectly for users with custom browser settings for large fonts.

## Testing done

Open chrome settings and set the font size to Large and then return to My HealtheVet Landing page and refresh browser.
Screenshot 2024-08-02 at 6 21 10 PM

Open Safari and set the font size to 24 and then return to My HealtheVet Landing page and refresh browser.
Screenshot 2024-08-02 at 6 21 43 PM

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Desktop |  Safari: Screenshot 2024-08-02 at 6 23 41 PM Chrome: Screenshot 2024-08-02 at 6 23 30 PM |  Safari: Screenshot 2024-08-02 at 6 16 58 PM Chrome: Screenshot 2024-08-02 at 6 16 26 PM |

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by Andrew565** [comment]
Contains failing tests, choosing not to review at this time. - T2 Support

File: src/platform/mhv/secondary-nav/components/MhvSecondaryNavMenu.jsx [hunk 1]
```diff
@@ -13,6 +14,84 @@ import MhvSecondaryNavItem from './MhvSecondaryNavItem';
...
+            resizeObserver.disconnect();
+          }
+          window.removeEventListener('resize', checkAndUpdateFontSize);
+        } catch (error) {
+          console.log('Error in cleanup function:', error.message);
+        }
+      };
+    },
+    [checkFontSize],
+  ); // Re-run effect if checkFontSize changes
```
Comment (radavis) [inline src/platform/mhv/secondary-nav/components/MhvSecondaryNavMenu.jsx:94]: Would it be possible to extract this functionality into a function/package and add tests?

Comment (radavis) [reply src/platform/mhv/secondary-nav/components/MhvSecondaryNavMenu.jsx:94]: > Would it be possible to extract this functionality into a function/package and add tests?

Sorry. To clarify, I'm stating this as a possible next step for this effort, and not requesting that you @jzucadi specifically should do this, unless you want to.

Comment (jzucadi) [reply src/platform/mhv/secondary-nav/components/MhvSecondaryNavMenu.jsx:94]: @radavis you are 100% right, this should be all extracted out of the file itself and include some tests. I did overlook this, let me get moving on that. Thank you for pointing this out!

Comment (jzucadi) [reply src/platform/mhv/secondary-nav/components/MhvSecondaryNavMenu.jsx:94]: @radavis heads up, I was not able to get to this.

**Pull request closed without merging**